### PR TITLE
Should not sync the volume when  claim.Spec.VolumeName is null 

### PR DESCRIPTION
**Pull request by houjun41544**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by jsafrane** [comment]
Does this fix any real bug? How can it be reproduced? To me it looks like the worst thing that can happen is that "error getting name of volume %q to get volume from informer" is logged at level 4.

**Comment by houjun41544** [comment]
@jsafrane Yes, this may not pruduce a bug except a log.  But would it be better to prevent a null volume from adding to volumequeue？

File: pkg/controller/volume/persistentvolume/pv_controller_base.go [hunk 1]
```diff
@@ -243,6 +243,10 @@ func (ctrl *PersistentVolumeController) deleteClaim(claim *v1.PersistentVolumeCl
 	_ = ctrl.claims.Delete(claim)
 	glog.V(4).Infof("claim %q deleted", claimToClaimKey(claim))
 
+	if claim.Spec.VolumeName == "" {
```
Comment (jsafrane) [inline pkg/controller/volume/persistentvolume/pv_controller_base.go]: Please move this check below `volumeName := claim.Spec.VolumeName` and use `volumeName` in the check

Comment (verult) [reply pkg/controller/volume/persistentvolume/pv_controller_base.go]: Consider logging here as well.

**Comment by houjun41544** [comment]
@ jsafrane I have fixed it, Please review again, thanks.

**Pull request merged**